### ISSUE TITLE
[BUG] Java examples don't compile

### ISSUE THREAD

**Issue by Philzen** [OpenAPITools/openapi-generator#16701]
I have just tried updating a Spring Boot project from 4.3.1 to 7.0.1 and checked the example at:

https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator-maven-plugin/examples/spring.xml

but the project will no longer compile as i spews out a lot of `package io.swagger.v3.oas.annotations.* does not exist` messages.

I changed:

```xml
        <dependency>
            <groupId>io.swagger</groupId>
            <artifactId>swagger-annotations</artifactId>
            <version>1.5.8</version>
        </dependency>
```

to

```xml
        <dependency>
            <groupId>io.swagger.core.v3</groupId>
            <artifactId>swagger-annotations</artifactId>
            <version>2.2.16</version>
        </dependency>
```

and they compile again.

**Could somebody from the core team kindly advise that this is correct?**

I could then provide a PR to fix both the java-client and spring examples.

**Comment by Philzen** [comment]
Also i realized the java-client example uses jackson-databind-nullable@0.2.1 while the spring example still uses 0.1.0.

Shouldn't both be harmonized to use the latest 0.2.6 ?

**Comment by wing328** [comment]
>Shouldn't both be harmonized to use [the latest 0.2.6](https://github.com/OpenAPITools/jackson-databind-nullable/releases) ?

yes please

**Comment by Philzen** [comment]
@wing328 

See #16700 – i did some testing and found the examples actually were broken since v5.4, which made the switch to `io.swagger.core.v3`.

Could not test the springfox-swagger update as our projects don't need them, however from reading their documentation the 3.x version seems to be the way to go and compatible with the Spring Boot Version in the examples.

**Comment by wing328** [comment]
Just merged #16720 

Please pull the latest master to give it a try.
